### ISSUE TITLE
Compare pHashes of images

### ISSUE THREAD

**Issue by martinzavodny** [thomasheckmann/image-indexer#1]
Hey Thomas,

I tried your implementation of phash and it works well but I am curious whether there is solution to compare those hashes and evaluate whether two pictures are the same or not. For example the project of phash.org has this method and I dont know exactly how that works so I am little but stuck in the moment.

Thank you for answer in advance and have a nice day.

**Comment by thomasheckmann** [comment]
In ImageHashedType there are a couple of distance method to calculate the distance(degree of similarity) between to hash value, sound like what you are looking for.
/thomas

**Comment by martinzavodny** [comment]
So you are basically using plain hamming distance, nothing more special. Ok, as I mentioned I have been using library of phash.org until recently and the comparison of two almost similar pictures (actually it is one pic, just the copy of it has higher contrast ) returns hashes where their hamming distance is quite big, but still their method returns that the pics are similar.

After using your hash method I got even greater hamming distance of those pictures I mentioned, so I am curious how to evaluate those results to be able to conclude that the pictures are similar.

**Comment by thomasheckmann** [comment]
Yes, phash is all about hamming distance, phash.org does the same. Take a look at the test program, it computes the distance from images. You can just replace the test images with your own 

/thomas
